onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -294,6 +294,7 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
       // Calculate total input sizes for this operation.
       CalculateTotalInputSizes(&op_kernel_context, p_op_kernel,
                                input_activation_sizes, input_parameter_sizes, node_name_for_profiling);
+      concurrency::ThreadPool::StartProfiling(session_state.GetThreadPool());
```

Comment:
>concurrency::ThreadPool::StartPr [](start = 6, length = 32)

Would be nice to avoid this call all together if profiling is not enabled.

Reply:
I think this and StopProfiling are both guarded by is_profiling_enabled.  We should assert thought that the start/stop are properly matched to pick up any issues that creep in (i.e., stop only when started, and start only when stopped).

Reply:
It is under the condition - "is_profiler_enabled"

---
In reply to: [579429938](https://github.com/microsoft/onnxruntime/pull/6748#discussion_r579429938) [](ancestors = 579429938)

Reply:
Assertion added inside ThreadPoolProfiler - there will be exception if found any mismatch.

---
In reply to: [581166039](https://github.com/microsoft/onnxruntime/pull/6748#discussion_r581166039) [](ancestors = 581166039)

Reply:
Assertion added inside ThreadPoolProfiler - there will be exception if found any mismatch.

---
In reply to: [581166039](https://github.com/microsoft/onnxruntime/pull/6748#discussion_r581166039) [](ancestors = 581166039)